### PR TITLE
datetime, to and from string, tests

### PR DESCRIPTION
**Pull request by NeilMenne**

### CODE REVIEW

File: lib/hlclock/timestamp.ex [hunk 1]
```diff
@@ -97,6 +130,19 @@ defmodule HLClock.Timestamp do
     %T{time: t, counter: c, node_id: n}
   end
 
+  @doc """
+  Recover a Timestamp from the string representation.
+  """
+  def from_string(tstr) do
```
Comment (keathley) [inline lib/hlclock/timestamp.ex]: Why wouldn't we just use `decode/1` here?

Comment (keathley) [reply lib/hlclock/timestamp.ex]: Nevermind, I'm dumb.

File: lib/hlclock/timestamp.ex [hunk 2]
```diff
@@ -97,6 +130,19 @@ defmodule HLClock.Timestamp do
...
+    with {:ok, dt, _} <- tstr |> String.slice(0, 24) |> DateTime.from_iso8601(),
```
Comment (keathley) [inline lib/hlclock/timestamp.ex]: I think we can replace all the slices with a `[time, counter, node] = String.split("-")`

File: lib/hlclock/timestamp.ex [hunk 3]
```diff
@@ -97,6 +130,19 @@ defmodule HLClock.Timestamp do
     %T{time: t, counter: c, node_id: n}
   end
 
+  @doc """
+  Recover a Timestamp from the string representation.
```
Comment (keathley) [inline lib/hlclock/timestamp.ex]: This is really dumb but can we provide an example here so its clear what the output of this should look like?

File: lib/hlclock/timestamp.ex (hunk 2 above)
Comment (NeilMenne) [reply lib/hlclock/timestamp.ex]: You could with an extra step. Time would be three strings that would have to be rejoined.
`"2017-10-21T17:19:55.859Z-0000-0000000005474CE1" # ["2017", "10", "21T17:19:55.859Z", "0000", "0000000005474CE1"]`. Would it be better, in your opinion, to rejoin the time components this way or do the slice and dice?

**Pull request merged**